### PR TITLE
[Opt](multi-catalog) Opt split assignment to resolve uneven distribution.

### PR DESCRIPTION
**Pull request by kaka11chen**
## Proposed changes

[Opt] (multi-catalog) Opt split assignment to resolve uneven distribution. Currently only for `FileQueryScanNode`.

Referring to the implementation of Trino, 
- Local node soft affinity optimization. Prefer local replication node.
- Remote split will use the consistent hash algorithm is used when the file cache is turned on, and because of the possible unevenness of the consistent hash, the split is re-adjusted so that the maximum and minimum split numbers of hosts differ by at most `max_split_num_variance` split.
- Remote split will use the round-robin algorithm is used when the file cache is turned off.

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by kaka11chen** [comment]
run buildall

**Comment by kaka11chen** [comment]
run buildall

**Comment by kaka11chen** [comment]
run buildall

**Comment by kaka11chen** [comment]
run buildall

**Comment by kaka11chen** [comment]
run buildall

**Comment by kaka11chen** [comment]
run buildall

**Comment by kaka11chen** [comment]
run pipelinex_p0

File: fe/fe-common/src/main/java/org/apache/doris/common/Config.java [hunk 1]
```diff
@@ -2234,7 +2234,28 @@ public class Config extends ConfigBase {
             "When file cache is enabled, the number of virtual nodes of each node in the consistent hash algorithm. "
                     + "The larger the value, the more uniform the distribution of the hash algorithm, "
                     + "but it will increase the memory overhead."})
-    public static int virtual_node_number = 2048;
+    public static int virtual_node_number = 256;
+
+    @ConfField(mutable = true, description = {
+            "本地节点软亲缘性优化。尽可能地优先选取本地副本节点。",
+            "Local node soft affinity optimization. Prefer local replication node."})
+    public static boolean optimized_local_scheduling = true;
```
[the remark this answers is not in the document]
Comment (kaka11chen) [reply fe/fe-common/src/main/java/org/apache/doris/common/Config.java]: ok

**Comment by kaka11chen** [comment]
run buildall

**Comment by kaka11chen** [comment]
run buildall

**Review by morningman — APPROVED** [review]
LGTM

**Review by zddr — APPROVED** [review]

**Pull request merged**